packages/grant-explorer/src/features/api/rounds.ts

```diff
@@ -1,115 +1,29 @@
-  id: string;
-  chainId: ChainId;
-  createdAt: string;
-  roundMetaPtr: MetadataPointer;
-  applicationMetaPtr: MetadataPointer;
-  applicationsStartTime: string;
-  applicationsEndTime: string;
-  roundStartTime: string;
-  roundEndTime: string;
-  matchAmount: string;
-  token: string;
-  roundMetadata?: __deprecated_RoundMetadata;
-  projects?: { id: string }[];
-  payoutStrategy: {
-    id: string;
-    strategyName: RoundPayoutType;
-  };
-};
-
-export type __deprecated_RoundsQueryVariables = {
-  first?: number;
-  orderBy?:
-    | "createdAt"
-    | "matchAmount"
-    | "roundStartTime"
-    | "roundEndTime"
-    | "applicationsStartTime"
-    | "applicationsEndTime";
-  orderDirection?: "asc" | "desc";
-  where?: {
-    and: [
-      { or: __deprecated_TimestampVariables[] },
-      { payoutStrategy_?: { or: { strategyName: string }[] } },
-    ];
-  };
-};
-
-export type __deprecated_TimestampVariables = {
-  applicationsStartTime_lte?: string;
-  applicationsEndTime_gt?: string;
-  applicationsEndTime_lt?: string;
-  applicationsEndTime?: string;
-  applicationsEndTime_gte?: string;
-  roundStartTime_lt?: string;
-  roundStartTime_gt?: string;
-  roundEndTime_gt?: string;
-  roundEndTime_lt?: string;
-};
+import useSWR, { SWRResponse } from "swr";
+import { ChainId } from "common";
+import { createISOTimestamp } from "../discovery/utils/createRoundsStatusFilter";
+import { RoundGetRound, RoundsQueryVariables, useDataLayer } from "data-layer";
 
 export const useRounds = (
-  variables: __deprecated_RoundsQueryVariables,
+  variables: RoundsQueryVariables,
   chainIds: ChainId[]
-): SWRResponse<__deprecated_RoundOverview[]> => {
-  const { cache, mutate } = useSWRConfig();
+): SWRResponse<RoundGetRound[]> => {
   const dataLayer = useDataLayer();
 
-  const prewarmSwrCacheWithRoundsMetadata = async (
-    rounds: __deprecated_RoundOverview[]
-  ): Promise<void> => {
-    const roundsWithUncachedMetadata = rounds.filter(
-      (round) =>
-        cache.get(`@"metadata","${round.roundMetaPtr.pointer}",`) === undefined
-    );
-
-    const uncachedMetadata = await Promise.all(
-      roundsWithUncachedMetadata.map(async (round) => {
-        const cid = round.roundMetaPtr.pointer;
-        const metadata = await __deprecated_fetchFromIPFS(cid);
-        return [cid, metadata];
-      })
-    );
-
-    for (const [cid, metadata] of uncachedMetadata) {
-      mutate(["metadata", cid], metadata);
-    }
-
-    if (roundsWithUncachedMetadata.length > 0) {
-      // clear rounds cache
-      mutate(["rounds", chainIds, variables]);
-    }
-  };
-
   const query = useSWR(
     // Cache requests on chainIds and variables as keys (when these are the
     // same, cache will be used instead of new requests)
     ["rounds", chainIds, variables],
     async () => {
-      const { rounds } = await dataLayer.getLegacyRounds({
+      const spamRounds = await fetchSpamRounds();
+      const { rounds } = await dataLayer.getRounds({
```

Comment:
if we have both with await we can't fetch them concurrently. 
before it was starting the `getRounds` promise and then waiting for `fetchSpamRounds`. 
When both were done we were applying the filters. can we do it again here to speed it up?

Reply:
Good catch, thank you, fixed.